samples/arcgis-widget/src/components/Daylight/DaylightModel.ts

```diff
@@ -0,0 +1,36 @@
+import {
+    ComponentModelBase,
+    ComponentModelProperties,
+    importModel,
+    PropertyDefs,
+    serializable,
+} from "@vertigis/web/models";
+import { ItemType } from "@vertigis/arcgis-extensions/ItemType";
+
+export interface DaylightModelProperties extends ComponentModelProperties {
+    dateOrSeason?: "season" | "date";
+}
+
+@serializable
+export default class DaylightModel extends ComponentModelBase<
+    DaylightModelProperties
+> {
+    // TODO: Use correct type for `map` after 5.10 release.
+    @importModel(ItemType.MAP_EXTENSION)
+    map: any | undefined;
+
+    dateOrSeason: DaylightModelProperties["dateOrSeason"];
+
+    protected _getSerializableProperties(): PropertyDefs<
+        DaylightModelProperties
+    > {
+        const baseProperties = super._getSerializableProperties();
+        return {
+            ...baseProperties,
+            dateOrSeason: {
```

Comment:
Since you set up the serialization, I think that it would be helpful for the app config to explicitly have a value for this property, just to help connect the dots between the config and this model.

Reply:
Totally. Added `dateOrSeason: "season"` to the app.json.